src/librustc/traits/error_reporting.rs

```diff
@@ -1620,6 +1622,18 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                     err.note(&msg);
                 }
             }
+            ObligationCauseCode::BindingObligation(item_def_id, span) => {
+                let item_name = tcx.def_path_str(item_def_id);
+                let msg = format!("required by this bound in `{}`", item_name);
+                if let Some(ident) = tcx.opt_item_name(item_def_id) {
+                    err.span_label(ident.span, "");
+                }
+                if span != DUMMY_SP {
```

Comment:
This construct to use `.span_label` on having a span and `.note` otherwise seems to come up every now and then. I recently reviewed a PR of yours that did this. Consider adding a method on `err` that handles this?

Reply:
It's tricky because I don't have a feel for which case is more common for wanted behavior: silent fallback or note... We should end up with two `span_label` methods, one "with fallback note" and one "silent if DUMMY_SP". I would like to do that as part of a more concerted check throughout the compiler to clean up in one go.

Reply:
Fair, but please collect these notes in a new issue or otherwise there is no chance we'll ever get to it / remember.

Reply:
https://github.com/rust-lang/rust/issues/64184